### ISSUE TITLE
Resolution of #load'ed files doesn't work as I would expect when doing "Execute in Interactive"...

### ISSUE THREAD

**Issue by KevinH-MS** [dotnet/roslyn#8873]
1. Create a file "a.csx"
1. In the same folder, create a file "b.csx" that contains:
```
#load "a.csx"
```
1.  Open "b.csx" in Visual Studio
1.  Ctrl + A
1.  Alt + Enter

**RESULTS:**
```
(1,7): error CS1504: Source file 'a.csx' could not be opened -- Could not find file.
```
**EXPECTED RESULTS:**
No errors

**Comment by KevinH-MS** [comment]
FYI @ManishJayaswal @drognanar

**Comment by tmat** [comment]
We don't support editing .csx in VS atm.

**Comment by KevinH-MS** [comment]
I don't really understand the utility of "Execute in Interactive" then... :wink:

I get that we're not supporting a "full" editing experience, but it seems that basic colorization/intellisense for .csx is enabled now so I would expect this to work as well.

I think it's confusing to say that, "You can only execute things are valid statements in both C# and C# script".

**Comment by tmat** [comment]
Basic Intellisense yes, but the default #r's that are available when executing the script in csi for example aren't present. 

I guess we could change the working directory of the REPL to the directory containing the script, or set the search path to include the script directory and that shouldn't be to hard to implement.

**Comment by tmat** [comment]
I'm not sure we should actually do that though. You're sending code to interactive that's set up in a way the user wanted. It'd be odd to change the settings. 

You can't expect arbitrary code that you select to work in the REPL. If you don't send a function definition into the repl before you send it's call site it won't work. #load is similar - if you don't set your CWD or search path to include the script directory it won't work.

**Comment by KevinH-MS** [comment]
Yeah, I agree that arbitrary code may not work, but I was thinking we could at least somehow add the current script file's folder path to the resolver path for ```#load```.

**Comment by tmat** [comment]
Adding the folder might break other #load's already in the REPL though - not the submitted ones, they are already resolved, but the user might fetch one in the history that used to load a script from somewhere and now it loads it from elsewhere. That could be super confusing. F# would have the same problem, have you checked ho it works?

**Comment by KevinH-MS** [comment]
I was thinking it could be the last path we search, but good point, I'll try to see what F# does.

**Comment by drognanar** [comment]
F# actually does not have that problem since it does not add any submissions generated via `execute in interactive` to the interactive output and history. Also, it does resolve the paths against the script directory.

**Comment by tmat** [comment]
I'ts not about the submissions sent to the REPL, it's about the submissions typed manually in the REPL prior, for example,

```C#
> ReferencePaths
["C:\Foo"]
> #load "a.csx"
// loaded from C:\Foo\a.csx
```

Send
```
#load "b.csx"
```
from script ```C:\Bar\c.csx```

```C#
> #load "a.csx" (from history)
```

If we changed the CWD or ReferencePaths in any other way than adding ```C:\Bar``` to the and of ReferencePaths then repeated execution (invoked e.g. from history) of ```#load "a.csx"``` might have different effect.

Let's say we append the path to ReferencePaths. What if a script "C:\Foo\b.csx" exists. Then the send operation runs a different script than the user intended!

**Comment by tmat** [comment]
The only thing to do that doesn't lead to confusion is either 
1) do nothing and explain that we are just sending text to the repl, so it can change semantics if the repl is already in a state that affects the script, or
2) to rewrite the #load directive to an absolute path and send the rewritten snipped to the repl.

I think we should start with [1]. It's consistent with everything else. If you have, for example, already evaluated ```f(int x) => x + 1;``` in the repl and you have ```f(int x) => x * 2;``` in source and you send ```f(10);``` then it will return 11, not 20.

**Comment by KevinH-MS** [comment]
I would suggest that it's OK to have "surprising" results in a minority of cases if the 90+% case works as expected.  Today, the 90% case isn't working, and the work-around is tedious (especially if I have multiple scripts open in VS that don't live in the same directory).  I believe that appending the script directory to the list of resolver paths is a reasonable solution.  By default, this is the state of the REPL:
```
> ReferencePaths
SearchPaths { "C:\\Windows\\Microsoft.NET\\Framework\\v4.0.30319\\" }
> Environment.CurrentDirectory
"C:\\Users\\kevinh"
```
I would assert that it's very unlikely that you'll unexpectedly load a .csx from those locations.  If you go and customize your environment, then it should ***not*** be surprising that you're loading from the custom location (i.e. in practice, I don't think this will lead to much confusion at all).
